### ISSUE TITLE
Make subject_set.relation optional for the DELETE relationship endpoint

### ISSUE THREAD

**Issue by LucSalommez** [ory/keto#1329]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/keto/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/keto/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Network](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

TLDR: When we call the DELETE endpoint the parameter `relation` is optional but in a Subject Set it is mandatory which forces to hardcode all possible relations to make sure all tuples are deleted.

It is currently possible to delete relationships by calling the permission API and providing the values that should match the tuples we want to delete.

Let's say the following namespaces exist:
- User
- Group
- Company

It is possible to remove all User to Group relations for a specific group with this payload:
```json
{
    "namespace": "Group",
    "object": "groupID",
}
```

Here "relation" is optional, thus enabling to delete all the Group tuples even if the relations are different (for exemple "members", "managers" ...).

Now let's take a Subject Set example where `members` of `myGroup` are `employees` of `company` and `managers` of `myGroup` are `group_managers` of `company`

Imagine we want to delete `myGroup`, which would imply deleting all the tuples where `myGroup` is used as a Subject Set.
What I would like to do following the first example logic is to delete all tuples like this:

```json
{ 
    "namespace": "Company",
    "object": "company",
    "subject_set": {
        "namespace": "Group",
        "object": "myGroup"
    }
}
```

But when calling the endpoint I receive this error:
```
{
  "error": {
    "code": 400,
    "status": "Bad Request",
    "message": "incomplete subject, provide \"subject_id\" or a complete \"subject_set.*\""
  }
}
```

Which forces to define a "relation". This means that if the development team adds a new relation inside Group that is used as a Subject Set somewhere else, the backend code will have to be edited in the Group deletion process to also delete the new relations.

EDIT: I also tried with subject_set.relation = "" which returns a 204 and has no effects on my relations.

### Describe your ideal solution

It would be a bit more maintainable to make the "subject_set.relation" parameter optional and delete all tuples the same way as when not using a Subject Set.

### Workarounds or alternatives

Making a new request for each possible relation value.

### Version

v0.11.0

**Comment by LucSalommez** [comment]
I tried to compare the behavior with the PATCH endpoint.
Although the PATCH endpoint properly adds my tuples it does not behave the same way as the DELETE endpoint when trying to delete.

To take the same example as above I try to call the PATCH endpoint with this:
```json
[
	{
    "action": "delete",
    "relation_tuple": {
      "namespace": "Company",
      "object": "company",
      "subject_set": {
        "namespace": "Group",
        "object": "myGroup",
        "relation": "members"
      }
    }
  }
]
```

With this same payload:
- The DELETE endpoint delete all the tuples matching the values
- The PATCH endpoint deletes nothing

This is another issue but is a bit related so I posted it here.
Let me know If I should write a proper issue.
